crypto/s2n_libcrypto.c

```diff
@@ -35,17 +35,17 @@
  * the new API. When dropping OpenSSL 1.0.2 support, we can move to the new API.
  */
 
-/* Version name for OpenSSL depends on the version string, but for AWS-LC and
- * BoringSSL, this can be statically asserted.
+/* The result of SSLeay_version(SSLEAY_VERSION) for OpenSSL and AWS-LC depend on the
+ * version. AWS-LC and BoringSSL have consistent prefixes that can be statically asserted.
  *
  * https://github.com/awslabs/aws-lc/commit/8f184f5d69604cc4645bafec47c2d6d9929cb50f
  * has not been pushed to the fips branch of AWS-LC. In addition, we can't
  * distinguish AWS-LC fips and non-fips at pre-processing time since AWS-LC
  * doesn't distribute fips-specific header files.
  */
-#define EXPECTED_AWSLC_VERSION_NAME_FIPS_OR_OLD "BoringSSL"
-#define EXPECTED_AWSLC_VERSION_NAME_NON_FIPS    "AWS-LC"
-#define EXPECTED_BORINGSSL_VERSION_NAME         "BoringSSL"
+#define EXPECTED_AWSLC_NAME_PREFIX_FIPS_OR_OLD "BoringSSL"
+#define EXPECTED_AWSLC_NAME_PREFIX_NON_FIPS    "AWS-LC"
+#define EXPECTED_BORINGSSL_NAME_PREFIX         "BoringSSL"
```

Comment:
Nit: These are the names of the libraries, but aren't they the prefixes of the versions?
```suggestion
#define EXPECTED_AWSLC_VERSION_PREFIX_FIPS_OR_OLD "BoringSSL"
#define EXPECTED_AWSLC_VERSION_PREFIX_NON_FIPS    "AWS-LC"
#define EXPECTED_BORINGSSL_VERSION_PREFIX         "BoringSSL"
```
And then the method below would be s2n_libcrypto_validate_expected_version_prefix